### PR TITLE
fix: detekt rules and code smells

### LINKED ISSUE #1044: Detekt rules disabled
## Use case

Some important rules for keeping the code clean were disabled, such as `InvalidPackageDeclaration` and `UnusedImports`.

## Proposal

Enable the rules described above and adjust the project code as they require.

### PR DESCRIPTION
**Pull request by matheusribeirozup**
Signed-off-by: Matheus Ribeiro <matheus.ribeiro@zup.com.br>

### Related Issues

Closes #1044 

### Description and Example

This PR updates some Beagle dependencies like detekt and enables some of its rules.

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/6b0b7b032e88ae2b2454ba534f53bc018c34f962/android/beagle/src/main/java/br/com/zup/beagle/android/utils/ToolbarManager.kt#L224">android/beagle/src/main/java/br/com/zup/beagle/android/utils/ToolbarManager.kt#L224</a> - Prefer “<code>SHOW_AS_ACTION_IF_ROOM</code>” instead of “<code>SHOW_AS_ACTION_ALWAYS</code>”</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          13 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/BuilderKotlinDslScreenBuilder.kt#L20">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/BuilderKotlinDslScreenBuilder.kt#L20</a> - The import ‘br.com.zup.beagle.ext.style’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/BuilderKotlinDslScreenBuilder.kt#L25">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/BuilderKotlinDslScreenBuilder.kt#L25</a> - The import ‘br.com.zup.beagle.widget.core.AlignItems’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ClickActionScreenBuilder.kt#L26">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/ClickActionScreenBuilder.kt#L26</a> - The import ‘br.com.zup.beagle.widget.layout.ScreenBuilder’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt#L21">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/NavigateActionsScreenBuilder.kt#L21</a> - The import ‘br.com.zup.beagle.ext.setId’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L25">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/builders/TextScreenBuilder.kt#L25</a> - The import ‘br.com.zup.beagle.widget.context.Bind’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/config/CorsConfig.kt#L19">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/config/CorsConfig.kt#L19</a> - The import ‘org.springframework.context.annotation.Bean’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/ColorConstants.kt#L1">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/ColorConstants.kt#L1</a> - The file does not contain a package declaration.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/ImageConstants.kt#L1">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/ImageConstants.kt#L1</a> - The file does not contain a package declaration.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/PathConstants.kt#L1">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/PathConstants.kt#L1</a> - The file does not contain a package declaration.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/StyleConstants.kt#L1">backend/automated-tests/src/main/kotlin/br/com/zup/beagle/automatedtests/constants/StyleConstants.kt#L1</a> - The file does not contain a package declaration.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/framework/src/main/kotlin/br/com/zup/beagle/constants/CorsConstants.kt#L20">backend/framework/src/main/kotlin/br/com/zup/beagle/constants/CorsConstants.kt#L20</a> - The import ‘br.com.zup.beagle.platform.BeaglePlatformUtil’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/widgets-dsl/src/main/kotlin/br/com/zup/beagle/ui/ImageBuilder.kt#L20">backend/widgets-dsl/src/main/kotlin/br/com/zup/beagle/ui/ImageBuilder.kt#L20</a> - The import ‘br.com.zup.beagle.widget.context.Bind’ is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/90fbe8bfc3d718e9b85f0e310e59f92fe563b13a/backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/form/InputWidget.kt#L20">backend/widgets/src/main/kotlin/br/com/zup/beagle/widget/form/InputWidget.kt#L20</a> - The import ‘br.com.zup.beagle.widget.action.Action’ is unused.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_backend">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: config/detekt-default-config.yml [hunk 1]
```diff
@@ -501,6 +511,8 @@ potential-bugs:
 
 style:
   active: true
+  ClassOrdering:
+    active: false
```
Comment (uziasferreirazup) [inline config/detekt-default-config.yml:515]: this function is very nice I think you can enable another PR with refactoring to adjust the code.

**Review by uziasferreirazup — APPROVED** [review]

**Review by viniciusguardieirozup — APPROVED** [review]

**Pull request merged**